lib/oprt.gi

```diff
@@ -997,7 +997,7 @@ local orb,d,gen,i,p;
   for p in orb do
     for gen in acts do
       i:=act(p,gen);
-      MakeImmutable(i);
+      if not IsHPCGAP then MakeImmutable(i);fi;
```

Comment:
Huh, I wondered about this and then found this in a commit message:
> The HPCGAP test otherwise gives a cryptic error message:
> Error, MakeImmutable: Argument has inaccessible subobjects

So that message means that subobject of `i` was either created in a different thread, or for some other reason the current thread has no access to it. Since I can't see from just looking at this what kind of object `i` is when the error arose, it's a bit difficult to analyze further. But in any case, not calling `MakeImmutable` likely just defers the problem. Would be better to get to the root cause.

Is there a simple input one can use to trigger that error (after changing this line and/or the other back) ?

Reply:
(Just to clarify: I am not expecting @hulpke to investigate this HPC-GAP specific issue, but perhaps @ChrisJefferson or me could have a look, once we know how to trigger it, so I am hoping for a hint on that by @hulpke)

Reply:
@fingolfin It is tests in the standard test suite using automorphism group calculations. I have removed the commit again, so the test will show them up. (I will update this comment later with teh concrete command triggering the error.)
The change triggering the error message is an orbit algorithm on objects that are lists , one of whose entries is a conjugacy class.

Reply:
(ping @fingolfin) Any update on this? I would like to get this chunk of changes merged to not have too much of a discrepancy between my main branch and work branch. Could we keep the HPCGAP workaround for merging and then have the debugging in HPCGAP?

Reply:
Sorry, I missed your reply :-(.

I had a look. In the quick way to reproduce (thanks, that helped a lot!), the variable `i` is a plain list containing permutations. The problem is that this list is in the "public region", i.e. all threads can *read* it -- but that means we are not allowed to write to it, and we need to write to it to make it immutable (even though that really just means flipping a bit here).

However, it's really odd that a mutable plain list is in the public region. That seems like a bug to me. I'll try to figure out where those come from, it's odd.

Reply:
Tracked it down, fix is in https://github.com/gap-system/gap/pull/3848. This is a really old bug which just started to have a more visible bad effect in HPC-GAP. All in all, I am actually glad you run into this and we persisted in digging it out :-). Sorry for taking so long to complete this, though.

CC @rbehrends @ChrisJefferson

Reply:
@fingolfin Thank you for the fix. Yes, its good this bug was found.